tests/caps_word/caps_word_unicodemap/test_caps_word_unicodemap.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 Google LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "keyboard_report_util.hpp"
+#include "keycode.h"
+#include "test_common.hpp"
+#include "test_fixture.hpp"
+#include "test_keymap_key.hpp"
+
+using ::testing::_;
+using ::testing::AnyNumber;
+using ::testing::AnyOf;
+using ::testing::InSequence;
+
+extern "C" {
+enum unicode_names {
+    ENDASH,
+    EMDASH,
+    DELTA_LOWERCASE,
+    DELTA_UPPERCASE,
+};
+
+const uint32_t unicode_map[] PROGMEM = {
+    [ENDASH]          = 0x2013,
+    [EMDASH]          = 0x2014,
+    [DELTA_LOWERCASE] = 0x03b4,
+    [DELTA_UPPERCASE] = 0x0394,
+};
+
+#define U_DASH XP(ENDASH, EMDASH)
+#define U_DELTA XP(DELTA_LOWERCASE, DELTA_UPPERCASE)
+
+bool caps_word_press_user(uint16_t keycode) {
+    switch (keycode) {
+        // Keycodes that continue Caps Word, with shift applied.
+        case U_DELTA:
+            add_weak_mods(MOD_BIT(KC_LSFT)); // Apply shift to next key.
+            return true;
+
+        // Keycodes that continue Caps Word, without shifting.
+        case U_DASH:
+            return true;
+
+        default:
+            return false; // Deactivate Caps Word.
+    }
+}
+} // extern "C"
+
+namespace {
+// Sets gmock expectation that the Unicode symbol `unicode_map[index]` is sent
+// with UC_LNX input mode. For instance for U+2013, the keys sent are:
+// "Ctrl+Shift+U, 2, 0, 1, 3, space".
+//
+// NOTE: This function should be called within an `InSequence` context.
+void expect_unicodemap_key(TestDriver* driver, int index) {
+    EXPECT_CALL(*driver, send_keyboard_mock(KeyboardReport(KC_LCTL, KC_LSFT, KC_U)));
+
+    const uint16_t code = unicode_map[index];
+    // clang-format off
+    static const uint8_t hex_keycodes[] = {
+        KC_0, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7,
+        KC_8, KC_9, KC_A, KC_B, KC_C, KC_D, KC_E, KC_F
+    };
+    // clang-format on
+    for (int i = 12; i >= 0; i -= 4) {
+        // Extract hex digit and convert to keycode.
+        const uint8_t keycode = hex_keycodes[(code >> i) & 0xf];
+        EXPECT_CALL(*driver, send_keyboard_mock(KeyboardReport(keycode)));
+    }
+
+    EXPECT_CALL(*driver, send_keyboard_mock(KeyboardReport(KC_SPC)));
+}
+} // namespace
```

Comment:
Can this function be used universally by other tests that exercise Unicode emulation? If so this should become part of `TestDriver` and probably (?) be written in all caps to follow the Gtest convention on assertion macros.

Reply:
Done. Added an `EXPECT_UNICODE` macro that sets a gmock expectation that a Unicode code point will be sent.

While I was at it, I also added an `EXPECT_REPORT` macro for a shorthand notation to expect a given keyboard report, since this is so frequent in the tests.